native/packages/react-native-bpk-component-horizontal-nav/src/BpkHorizontalNav.js

```diff
@@ -42,39 +43,90 @@ const styles = StyleSheet.create({
     flexGrow: 1,
     justifyContent: 'space-around',
   },
+  selectedIndicator: {
+    backgroundColor: tokens.colorBlue700,
+    height: tokens.borderSizeLg,
+    width: 50,
+  },
 });
 
-const BpkHorizontalNav = (props) => {
-  const {
-    children,
-    spaceAround,
-    style,
-    ...rest
-  } = props;
-  const navStyle = [styles.horizontalNav];
-  if (spaceAround) {
-    navStyle.push(styles.spaceAround);
+class BpkHorizontalNav extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isReady: false };
+    this.onChildLayout = this.onChildLayout.bind(this);
   }
-  return (
-    <ScrollView
-      alwaysBounceHorizontal={false}
-      contentContainerStyle={[navStyle, style]}
-      horizontal
-      showsHorizontalScrollIndicator={false}
-      {...rest}
-    >
-      { children }
-    </ScrollView>
-  );
-};
+  onChildLayout = (event) => {
+    const { width, x } = event.nativeEvent.layout;
+    this.childrenPositions.push({ width, x });
+    this.checkChildrenPositions(React.Children.count(this.props.children), this.childrenPositions.length);
+  };
+  checkChildrenPositions(childrenLength, renderedChildren) {
+    if (childrenLength === renderedChildren) {
+      // onLayout is async, so sort the children to ensure correct order.
+      this.childrenPositions = this.childrenPositions.sort((a, b) => a.x > b.x);
```

Comment:
probably  different name rather than `a` and `b` will make it easier to understand

Reply:
Hard to say - we could use `obj1` and `obj2` but they might not be any more descriptive. I'm struggling to think of something that's more descriptive than `a` and `b` for such a trivial function.